test/javascript_prelude/main.mjs

```diff
@@ -256,6 +256,15 @@ let weak_set = new WeakSet([map, set]);
 assertEqual(weak_set, weak_set);
 assertNotEqual(new WeakSet([map, set]), new WeakSet([map, set]));
 
+// RegExp are not equal unless they have reference equality
+let re = new RegExp("/test/g");
+let re_literal = /test/g;
+assertEqual(re, re);
+assertEqual(re_literal, re_literal);
+assertNotEqual(/test/g, /test/g);
+assertNotEqual(new RegExp("/test/g"), new RegExp("/test/g"));
+assertNotEqual(re, re_literal);
```

Comment:
Why are these not equal? They have the same patterns and the same flags.

Reply:
See: https://github.com/gleam-lang/gleam/pull/2479#discussion_r1435174456